include/deal.II/sundials/arkode.h

```diff
@@ -786,6 +907,319 @@ namespace SUNDIALS
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an
+     * assertion will be thrown.
+     */
+    std::function<int(double t, const VectorType &v, VectorType &Mv)>
+      mass_times_vector;
+
+    /**
+     * A function object that users may supply and that is intended to setup
+     * the mass matrix. This function is called by ARKode any time a mass
+     * matrix update is required. The user should compute the mass matrix (or
+     * update all the variables that allow the application of the mass matrix).
+     * This function is guaranteed to be called by ARKode at least once, before
+     * any call to mass_times_vector().
+     *
+     * ARKode supports the case where the mass matrix may depend on time, but
+     * not the case where the mass matrix depends on the solution itself.
+     *
+     * If the user does not provide a mass_times_vector() function, then the
+     * identity is used. If the mass_times_setup() function is not provided,
+     * then mass_times_vector() should do all the work by itself.
+     *
+     * If the user uses a matrix-based computation of the mass matrix, then
+     * this is the right place where an assembly routine should be called to
+     * assemble the matrix. Subsequent calls (possibly  more than one) to
+     * mass_times_vector() can assume that this function has been called at
+     * least once.
+     *
+     * @note No assumption is made by this interface on what the user
+     * should do in this function. ARKode only assumes that after a call to
+     * mass_times_setup() it is possible to call mass_times_vector().
+     *
+     * @param t the current evaluation time
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (ARKodeReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an
+     * assertion will be thrown.
+     */
+    std::function<int(const double t)> mass_times_setup;
+
+    /**
+     * A function object that user may supply and that is intended to compute
+     * the product of the Jacobian matrix with a given vector `v`. The Jacobian
+     * here refers to $J=\frac{\partial f_I}{\partial y}$, i.e., the Jacobian of
+     * the user-specified implicit_function.
+     *
+     * A call to this function should store in `Jv` the result of $J$
+     * applied to `v`.
+     *
+     * Arguments to the function are
+     *
+     * @param[in] v  the vector to be mulitplied by the Jacobian
+     * @param[out] Jv the vector to be filled with the product J*v
+     * @param[in] t  the current time
+     * @param[in] y  is the current $y$ vector for the current ARKode internal
+     * step
+     * @param[in] fy  is the current value of the implicit right-hand side at y,
+     * $f_I (t_n, y)$.
```

Comment:
Formatting.

Reply:
@peterrum What exactly should I change here? Indent the second line? I looked through a couple files and couldn't see any inconsistency here.